src/base.js

```diff
@@ -122,6 +123,28 @@ class Base extends EventEmitter {
     return nsContext;
   }
 
+  addSoapHeadersToEnvelope(soapHeaderElement, xmlHandler) {
+    for (let i = 0, n = this.soapHeaders.length; i < n; i++) {
+      let soapHeader = this.soapHeaders[i];
+      let elementDescriptor;
+      if (typeof soapHeader.value === 'object') {
+        if (soapHeader.qname !== undefined) {
+          if (soapHeader.qname.nsURI !== undefined) {
+            let element = this.findElement(soapHeader.qname.nsURI, soapHeader.qname.name);
+            elementDescriptor =
+              element && element.describe(this.wsdl.definitions);
+          }
+        }
+        xmlHandler.jsonToXml(soapHeaderElement, null, elementDescriptor,
+          soapHeader.value);
+      } else { //soapHeader has XML value
+        XMLHandler.parseXml(soapHeaderElement, soapHeader.xml);
+        xmlHandler.jsonToXml(soapHeaderElement, null, null,
```

Comment:
The parseXml already adds the xml string into the soapHeaderElement. I don't see the need to call xmlHandler.jsonToXml again unless we allow both xml and value.